### PR TITLE
Documentation for release 2.3 and Doxygen links

### PR DESCRIPTION
**Pull request by ccadar**

### CODE REVIEW

File: _includes/footer.html [hunk 1]
```diff
@@ -12,7 +12,7 @@
           <a class="sitemap--list__link sitemap--list__link--parent" href="https://twitter.com/kleesymex">Twitter</a>
         </li>
         <li class="sitemap__item sitemap__item--doxygen">
-          <a class="sitemap--list__link sitemap--list__link--parent" href="">Doxygen</a>
+          <a class="sitemap--list__link sitemap--list__link--parent" href="https://klee.github.io/releases/docs/v2.3/doxygen/html">Doxygen</a>
```
Comment (251) [inline _includes/footer.html:15]: Shouldn't it be: `href="{{ site.baseurl }}/releases/docs/v2.3/doxygen/html"`?

Comment (ccadar) [reply _includes/footer.html:15]: Good catch, but no.  This is what I had initially, but `site.baseurl` is set to `https://klee.github.io/releases/docs/v2.3/` for the 2.3 release so the link would be incorrect. 
It's obviously not a very elegant solution, but I think the right one given the special handling of Doxygen now.

**Comment by ccadar** [comment]
> I think you should add the Doxygen-link before generating the page, otherwise the link in the footer for release 2.3 is dead. 

I think it all works now, but I had to make some manual adjustments.

> Do you really want to add Doxygen documentation for all future releases? Maybe just generating it for master from time to time is sufficient?

Not sure.  I feel it's nice to have it for every release, but I don't feel strongly about it.  Are you concerned about the number of files involved?

**Comment by ccadar** [comment]
For Git or the website?

**Comment by ccadar** [comment]
Let me know if you have particular experiences/stats about this (I don't)

**Comment by ccadar** [comment]
Basically, I was thinking that we would only add new Doxygen pages for new releases, which wouldn't be more than twice a year.   (For the master branch, we could update them in place, if the updates are incremental, or we would host them somewhere else)

**Comment by 251** [comment]
But its ~9000 new files per release... If anyone is interested in the release-specific documentation it is still easy to generate from the sources.

File: _includes/footer.html (hunk 1 above)
Comment (251) [reply _includes/footer.html:15]: So: `href="{{ site.baseurl }/doxygen/html"`?

**Comment by ccadar** [comment]
So what's your proposal?  A global doxygen directory which we update at every release?

File: _includes/footer.html (hunk 1 above)
Comment (ccadar) [reply _includes/footer.html:15]: But this won't work for master then.

**Comment by 251** [comment]
A global doxygen that we update occasionally until someone integrates it with upstream merges.

**Comment by ccadar** [comment]
OK, I thought a bit more about this, and a global Doxygen should suffice.

**Pull request closed without merging**